elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -782,56 +781,126 @@ public boolean isIdGenerated() {
-            // reflect reality (i.e. if a user is creating an object with values, he/she knows those values
-            // already).
-            skipNew = true;
         } else {
-            Class<?> idType = dictionary.getIdType(entityType);
-            Object idVal = CoerceUtil.coerce(id, idType);
-            String idField = dictionary.getIdFieldName(entityType);
-
-            filterExpression = Optional.of(new FilterPredicate(
-                    new FilterPredicate.PathElement(
-                            entityType,
-                            idType,
-                            idField),
-                    Operator.IN,
-                    Collections.singletonList(idVal)));
-        }
-
-        Set<PersistentResource> resources =
-                filter(ReadPermission.class,
-                        (Set) getRelationChecked(relation, filterExpression, Optional.empty(), Optional.empty()),
-                        skipNew);
-
-        for (PersistentResource childResource : resources) {
-            if (childResource.matchesId(id)) {
-                return childResource;
+            if (!ids.isEmpty()) {
+                String typeAlias = dictionary.getJsonAliasFor(entityType);
+                Map<String, PersistentResource> newResourceMap = requestScope.getNewPersistentResources().stream()
+                        .filter(r -> typeAlias.equals(r.getType()))
+                        .collect(Collectors.toMap(r -> (String) r.getUUID().orElse(""), r -> r));
+
+                // Fetch our set of new resources that we know about since we can't find them in the datastore
+                newResources = new LinkedHashSet<>(ids.stream()
+                        .filter(newResourceMap::containsKey)
+                        .map(newResourceMap::get)
+                        .collect(Collectors.toList()));
+
+                FilterExpression idExpression = buildIdFilterExpression(ids, entityType);
+
+                // Combine filters if necessary
+                filterExpression = filter
+                        .map(fe -> joinFilterExpressions(idExpression, fe))
+                        .orElse(idExpression);
+            } else {
+                filterExpression = filter.orElse(null);
             }
         }
-        throw new InvalidObjectIdentifierException(id, relation);
+
+        // TODO: Filter on new resources?
+        // TODO: Update pagination to subtract the number of new resources created?
+
+        Set<PersistentResource> existingResources = filter(ReadPermission.class,
+            (Set) getRelationChecked(relation, Optional.ofNullable(filterExpression), sorting, pagination));
+
+        // TODO: Sort again in memory now that two sets are glommed together?
+
+        Set<PersistentResource> allResources = Sets.union(newResources, existingResources);
+
+        String missedIds = ids.stream()
+                .filter(id -> allResources.stream().noneMatch(p -> p.matchesId(id)))
```

Comment:
Simpler to do:
```
List<String> allIds = allResources.stream().map(PersistentResoruce::getUUID())
List<String> missingIds = Sets.difference(ids, allIds)
if (!missedIds.isEmpty()) {
   throw new InvalidObjectIdentifierException(missedIds, relation);
}
```

Reply:
Still a problem here: `matchesId` does more than a simple comparison against UUID. Since this list isn't _exclusively_ UUID's, I think this would likely be wrong.